holidays/holiday_base.py

```diff
@@ -223,6 +238,7 @@ def __init__(
         prov: Optional[str] = None,  # Deprecated.
         state: Optional[str] = None,  # Deprecated.
         language: Optional[str] = None,
+        categories: Optional[Set[str]] = None,
```

Comment:
However, this one is potentially a bad choice:
```python
>>> import holidays
>>> categories = {holidays.constants.PUBLIC}
>>> us = holidays.US(categories=categories)
>>> ua = holidays.UA(categories=categories)
>>> us.categories
{'public'}
>>> ua.categories
{'public'}
>>> us.categories.add(holidays.constants.BANK)
>>> us.categories
{'public', 'bank'}
>>> ua.categories
{'public', 'bank'}
```

I suggest keeping a Tuple here.